### PR TITLE
External Languages ( C )

### PR DESCRIPTION
**Pull request by IshaanAdarsh**
The Quick Start aims to guide users through the process of creating a simple PostgreSQL extension using c.

### CODE REVIEW

File: Quick-Start [DA]/pg_cext.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+}
+```
+
+### Step 5: Create the SQL Script file:
+
+Create a control file named `pg_cext--1.0.0.sql` with the following content:
+
+```sql
+CREATE OR REPLACE FUNCTION
+addme(int,int) RETURNS int AS 'MODULE_PATHNAME','addme'
```
Comment (IshaanAdarsh) [inline Quick-Start [DA]/pg_cext.md]:
```suggestion
add_nums(int,int) RETURNS int AS 'MODULE_PATHNAME','add_nums'
```

File: Quick-Start [DA]/pg_cext.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
+# External Languages in PostgreSQL:
+External languages in PostgreSQL provide a powerful way to integrate additional programming languages, extending the database's capabilities beyond its native support. Developers can use these languages to implement custom functionality, access external libraries, and leverage their expertise in preferred programming languages.
+
+Using external languages offers several advantages:
+
+1. **Language Flexibility:** Integrate a wide range of languages like Python, Ruby, or Java to perform tasks that may not be as straightforward in SQL.
```
Comment (theory) [inline Quick-Start [DA]/pg_cext.md:6]: I think Python, Ruby, and Java are all PLs, not external languages. Not sure what's really supported aside from C and maybe C++; whatever can generate a dynamic library, I would guess.

File: Quick-Start [DA]/pg_cext.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+2. **Code Reusability:** Reuse existing code written in preferred languages, reducing development time and effort.
+
+3. **Performance Optimization:** Choose languages with performance advantages for specific tasks, improving the overall extension efficiency.
```
Comment (theory) [inline Quick-Start [DA]/pg_cext.md:10]: I think these are much more about PLs, but *execution* performance is definitely a big deal for C extensions.

File: Quick-Start [DA]/pg_cext.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+4. **Ecosystem Integration:** Tap into well-established language ecosystems, leveraging community-contributed libraries and best practices.
+
+To utilize external languages in a PostgreSQL extension, follow these steps:
+
+1. **Implement the Language:** Develop the code and functionality to support the external language within the extension.
```
Comment (theory) [inline Quick-Start [DA]/pg_cext.md:16]: I don't follow this. I don't have to implement C to build a C extension. A little confused.

File: Quick-Start [DA]/pg_cext.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+# Quick Start: Creating a PostgreSQL Extension using C as an External Language
+
+In this quick start guide, we will create a simple PostgreSQL extension using C as an external language. The extension will include a function to add two numbers.
+
+## Prerequisites
+
+- [PostgreSQL](https://www.postgresql.org/download/) installed and running on your system.
+- Basic knowledge of C programming.
+- Familiarity with the PostgreSQL extension development process.
```
Comment (theory) [inline Quick-Start [DA]/pg_cext.md:32]: Also `make`, a compiler, and the Postgres dev libraries.

File: Quick-Start [DA]/pg_cext.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+1. **.o Files (Object Files):** During the build process, the source code of the extension (e.g., `my_extension.c`) is translated into machine-readable code represented by object files. These files contain the compiled code for individual functions or components of the extension. However, they are not yet directly executable on their own.
+
+2. **.so Files (Shared Object Files):** The next step in the build process is linking the object files together to create a shared object file (`.so`). This file is a dynamically linked library that contains all the compiled code from the object files. It is also referred to as a shared library or DLL (Dynamic Link Library) on some systems. The `.so` file is the final form of the extension that PostgreSQL can use.
+
+**PGXS (PostgreSQL Extension Build Infrastructure):** 
+The PostgreSQL Extension Build Infrastructure (PGXS) is a set of macros and rules that simplify the build process for PostgreSQL extensions. It automates common build tasks and allows developers to use predefined variables and targets in their `Makefile`. By including PGXS in the `Makefile`, developers can leverage its functionality and ensure a standardized build process.
```
Comment (theory) [inline Quick-Start [DA]/pg_cext.md]:
```suggestion
The PostgreSQL Extension Build Infrastructure (PGXS) is a set of Makefile macros and rules that simplify the build process for PostgreSQL extensions. It automates common build tasks and allows developers to use predefined variables and targets in their `Makefile`. By including PGXS in the `Makefile`, developers can leverage its functionality and ensure a standardized build process.
```

**Review by theory — APPROVED** [review]
Just a few tweaks.

File: Quick-Start [DA]/pg_cext.md (hunk 2 above)
Comment (IshaanAdarsh) [reply Quick-Start [DA]/pg_cext.md:6]: yup your right, but just mentioning C and C++ doesn't really add value. 
Maybe could add languages like R or C#. Could you suggest some languages apart from this?

File: Quick-Start [DA]/pg_cext.md (hunk 4 above)
Comment (IshaanAdarsh) [reply Quick-Start [DA]/pg_cext.md:16]: I did create a simple C file to implement the sum function, this point is to just showcase that point.

File: Quick-Start [DA]/pg_cext.md (hunk 5 above)
Comment (IshaanAdarsh) [inline Quick-Start [DA]/pg_cext.md:32]:
```suggestion
- Familiarity with the PostgreSQL extension development process.
- Prerequisites:

- [PostgreSQL](https://www.postgresql.org/download/) installed and running on your system.
- Basic knowledge of C programming.
- Familiarity with the PostgreSQL extension development process.
- `make` utility for building and compiling software. 
- Compiler to convert your source code into executable machine code.
- Knowledge about the Postgres Development Libraries
```

Comment (IshaanAdarsh) [reply Quick-Start [DA]/pg_cext.md:32]: Added these points

**Pull request merged**